### ISSUE TITLE
ORPCA failing when normalize_poissonian_noise=True

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1557]
ORPCA is failing on my spectrum images when the argument `normalize_poissonian_noise` is set to `True` (which is the case a lot of the time in Electron Microscopy)
It is reporting a matrix multiplication shape mismatch. @tjof2 any thoughts?

A bit of a bodged hyperspy test as a working example:

```python
import hyperspy.api as hs
from hyperspy.tests.mva.test_rpca import TestORPCA
test = TestORPCA()
test.setup_method("ml") # why the setup_methods all take a "method" argument, I don't know
test.A.shape
(256, 1024)
s = hs.signals.Signal1D(test.A)
s.data[s.data < 0] = 0
s.decomposition(True, algorithm="ORPCA", output_dimension=5)

WARNING:hyperspy.learn.rpca:No method specified. Defaulting to 'CF' (closed-form solver)
WARNING:hyperspy.learn.rpca:No initialization specified. Defaulting to 'qr' initialization
WARNING:hyperspy.learn.rpca:Number of training samples for 'qr' method not specified. Defaulting to 10 samples
WARNING:hyperspy.learn.rpca:Nuclear norm regularization parameter is set to default: 1 / sqrt(nfeatures)
WARNING:hyperspy.learn.rpca:Sparse regularization parameter is set to default: 1 / sqrt(nfeatures)
/Users/thomas/miniconda3/envs/hs/lib/python3.5/site-packages/tqdm/_tqdm.py:65: DeprecationWarning: sys.getcheckinterval() and sys.setcheckinterval() are deprecated.  Use sys.setswitchinterval() instead.
  sys.setcheckinterval(100)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-38-1e32f72a728b> in <module>()
      6 s = hs.signals.Signal1D(A.A)
      7 s.data[s.data < 0] = 0
----> 8 s.decomposition(True, algorithm="ORPCA", output_dimension=5)

/Users/thomas/Dropbox/0_Git/hyperspy/hyperspy/learn/mva.py in decomposition(self, normalize_poissonian_noise, algorithm, output_dimension, centre, auto_transpose, navigation_mask, signal_mask, var_array, var_func, polyfit, reproject, return_info, **kwargs)
    434             # Rescale the results if the noise was normalized
    435             if normalize_poissonian_noise is True:
--> 436                 target.factors[:] *= self._root_bH.T
    437                 target.loadings[:] *= self._root_aG
    438 

ValueError: operands could not be broadcast together with shapes (256,5) (1024,1) (256,5) 
```

**Comment by tjof2** [comment]
@to266 did anything change when you modified this for dask? Never had this issue before.

**Comment by thomasaarholt** [comment]
Btw, I just fixed a typo in the code above. It doesn't change anything, however.

**Comment by to266** [comment]
PR'ed

The problem with this was that the old implementation iterated the "wrong" array axis to get samples, so in the class-based implementation I rewrote it do it properly (and close to identically for both an iterator and the numpy array). This means that the (old) function passes the transpose of the data to the class (which does the learning), which I forgot to include

**Comment by thomasaarholt** [comment]
Perhaps we can add some tests that operate on hyperspy signals and not just
np arrays as well? :)

**Comment by thomasaarholt** [comment]
Fixed in #1558 by @to266.
